lib/srv/db/cloud/watchers/redshift_serverless.go

```diff
@@ -0,0 +1,204 @@
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package watchers
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/redshiftserverless"
+	"github.com/aws/aws-sdk-go/service/redshiftserverless/redshiftserverlessiface"
+	"github.com/gravitational/trace"
+	"github.com/sirupsen/logrus"
+
+	"github.com/gravitational/teleport/api/types"
+	libcloudaws "github.com/gravitational/teleport/lib/cloud/aws"
+	"github.com/gravitational/teleport/lib/services"
+	"github.com/gravitational/teleport/lib/srv/db/common"
+)
+
+// redshiftServerlessFetcherConfig is the Redshift Serverless databases fetcher
+// configuration.
+type redshiftServerlessFetcherConfig struct {
+	// Labels is a selector to match cloud databases.
+	Labels types.Labels
+	// Region is the AWS region to query databases in.
+	Region string
+	// Client is the Redshift Serverless API client.
+	Client redshiftserverlessiface.RedshiftServerlessAPI
+}
+
+// CheckAndSetDefaults validates the config and sets defaults.
+func (c *redshiftServerlessFetcherConfig) CheckAndSetDefaults() error {
+	if len(c.Labels) == 0 {
+		return trace.BadParameter("missing parameter Labels")
+	}
+	if c.Region == "" {
+		return trace.BadParameter("missing parameter Region")
+	}
+	if c.Client == nil {
+		return trace.BadParameter("missing parameter Client")
+	}
+	return nil
+}
+
+// redshiftServerlessFetcher retrieves Redshift Serverless databases.
+type redshiftServerlessFetcher struct {
+	cfg redshiftServerlessFetcherConfig
+	log logrus.FieldLogger
+}
+
+// newRedshiftServerlessFetcher returns a new Redshift Serverless databases
+// fetcher instance.
+func newRedshiftServerlessFetcher(config redshiftServerlessFetcherConfig) (Fetcher, error) {
+	if err := config.CheckAndSetDefaults(); err != nil {
+		return nil, trace.Wrap(err)
+	}
+	return &redshiftServerlessFetcher{
+		cfg: config,
+		log: logrus.WithFields(logrus.Fields{
+			trace.Component: "watch:rss<", // (r)ed(s)hift (s)erver(<)less
+			"labels":        config.Labels,
+			"region":        config.Region,
+		}),
+	}, nil
+}
+
+// Get returns Redshift Serverless databases matching the watcher's selectors.
+func (f *redshiftServerlessFetcher) Get(ctx context.Context) (types.Databases, error) {
+	databases, workgroups, err := f.getDatabasesFromWorkgroups(ctx)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	if len(workgroups) > 0 {
+		vpcEndpointDatabases, err := f.getDatabasesFromVPCEndpoints(ctx, workgroups)
+		if err != nil {
+			if trace.IsAccessDenied(err) {
+				f.log.Debugf("No permission to get Redshift Serverless VPC endpoints: %v.", err)
+			} else {
+				f.log.Warnf("Failed to get Redshift Serverless VPC endpoints: %v.", err)
+			}
+		}
+
+		databases = append(databases, vpcEndpointDatabases...)
+	}
+	return filterDatabasesByLabels(databases, f.cfg.Labels, f.log), nil
```

Comment:
Thus VPC endpoints and workgroup (WG) endpoints provide a way for connecting to the same redshift serverless resource  I wonder if we should distinguish between them.

For instance let's consider setup, where teleport DB agent is deployed in the same VPC but also for redshift WG, has configured custom VPC endpoints to enable connectivity outside VPC for 3rd party tools. 
In that showing and discovering an endpoint seems to be superfluous.      

Another case is when a DB agent is deployed outside redshift VPC and a custom VPC endpoint was added to provide connectivity to the DB agent to redshift serverless resource. In that case, DB agent will also discover the workgroup endpoint but since DB agent is running in a different VPC WG endpoint will not be reachable. 

@greedy52 
Does it makes sense or I'm missing something how VPC endpoint and workgourp endpoint work ?

Reply:
I think that this is reasonable we can recommend to use either  `redshift-serverless` or  `redshift-serverless-vpc-endpoint` 
but also wonder about the scenario where there are many different endpoints but teleport db agent is deployed only inside one endpoint vpc. 
In that situation we will discovery all endpoint but only one will work. But maybe we can recommend to use `workgroup` setup ? The other option is to just check the connectivity to the endpoint url from the agent and filter it out if the endpoint is not accessible from the teleport db agent perspective (I think that we have talked about this before in different scenario )

Reply:
> In that situation, we will discover all endpoints but only one will work. But maybe we can recommend using a workgroup setup? 

Yes, I think the workgroup endpoint should be recommended. but at the same time, I believe usually they don't have a chance which vpc to deploy certain things otherwise they wouldn't have a VPC endpoint in the beginning.

> The other option is to just check the connectivity to the endpoint URL from the agent and filter it out if the endpoint is not accessible from the teleport DB agent perspective

I think we do want to have this "health-check" to see which URLs are reachable. With the introduction of discovery service though, the databases can be discovered by a different agent than the one that will do the connection. So I am leaning towards keeping the discovery dumb, and doing the connectivity check by the database agent separately (maybe send "health-check" info with DatabaseServer heartbeat). What do you think?

Another alternative that may help in this situation is to add a new toggle something like `same_vpc: true` so we only keep the ones in the same vpc.

Reply:
sounds good! I will make a small change today to this PR to allow specifying `aws-tags` through `teleport db configure` so we can recommend this without manually configuring the config file